tests/phpunit/tests/term/wpSetObjectTerms.php

```diff
@@ -430,4 +432,47 @@ public function test_non_existent_integers_should_be_ignored() {
 
 		$this->assertSame( array(), $tt_ids );
 	}
+
+	/**
+	 * Empty values should clear an object of all terms.
+	 *
+	 * @ticket 57923
+	 *
+	 * @dataProvider data_empty_value_should_clear_terms
+	 *
+	 * @param mixed $empty_value An empty.
+	 */
+	public function test_empty_value_should_clear_terms( $empty_value ) {
+		$post_id = self::$post_ids[0];
+
+		// Assign some terms.
+		wp_set_object_terms( $post_id, self::$term_ids, self::$taxonomy );
+
+		// Make sure the terms are set.
+		$terms = wp_get_object_terms( $post_id, self::$taxonomy, array( 'fields' => 'names' ) );
+		$this->assertNotEmpty( $terms );
+
+		// Remove terms by passing an empty value.
+		wp_set_object_terms( $post_id, $empty_value, self::$taxonomy );
+
+		// Make sure the terms have been removed.
+		$terms = wp_get_object_terms( $post_id, self::$taxonomy, array( 'fields' => 'names' ) );
+		$this->assertEmpty( $terms );
+	}
+
+	/**
+	 * Data provider for test_empty_value_should_clear_terms().
```

Comment:
```suggestion
	 * Data provider.
```

We're currently using this for data provider docblocks until consensus is reached on a consistent format.

```php
/**
 * Data provider.
 *
 * @return array[]
 */
```